### PR TITLE
PLT-436:Added support for credential type secret in oci registry

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

File: spectrocloud/resource_registry_oci_ecr.go [hunk 1]
```diff
@@ -129,6 +130,15 @@ func resourceRegistryEcrRead(ctx context.Context, d *schema.ResourceData, m inte
 		if err := d.Set("credentials", credentials); err != nil {
 			return diag.FromErr(err)
 		}
+	} else if registry.Spec.Credentials.CredentialType == models.V1AwsCloudAccountCredentialTypeSecret {
```
Comment (nikchern) [inline spectrocloud/resource_registry_oci_ecr.go:133]: can we use here switch statement please @SivaanandM would be more clear.

File: spectrocloud/resource_registry_oci_ecr_test.go [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+		"credential_type": "secret",
+		"secret_key":      "fasdfSADFsfasWQER23SADf23@",
+		"access_key":      "ASFFSDFWEQDFVXRTGWDFV",
+	}
+	credential = append(credential, cred)
+	d.Set("credentials", credential)
+	return d
+}
+
+func TestResourceRegistryEcrCreateSTS(t *testing.T) {
```
Comment (nikchern) [inline spectrocloud/resource_registry_oci_ecr_test.go:47]: this is great!

**Review by nikchern — CHANGES_REQUESTED** [review]

File: spectrocloud/resource_registry_oci_ecr_test.go (hunk 2 above)
Comment (SivaanandM) [reply spectrocloud/resource_registry_oci_ecr_test.go:47]: Thank you !

File: spectrocloud/resource_registry_oci_ecr.go (hunk 1 above)
Comment (SivaanandM) [reply spectrocloud/resource_registry_oci_ecr.go]: changed to switch case

File: go.mod [hunk 3]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.26.1
 	github.com/robfig/cron v1.2.0
 	github.com/spectrocloud/hapi v1.14.1-0.20230428134500-8a894264c61c
-	github.com/spectrocloud/palette-sdk-go v0.0.0-20230502044012-10512e17ec52
```
Comment (nikchern) [inline go.mod:13]: @SivaanandM could you please resolve conflicts?

**Review by nikchern — CHANGES_REQUESTED** [review]

File: go.mod (hunk 3 above)
Comment (SivaanandM) [reply go.mod]: resolved the conflict

**Review by nikchern — APPROVED** [review]

**Pull request merged**